assets/js/base/components/pagination/utils.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * Given the number of pages to display, the current page and the total pages,
+ * returns the min and max index of the pages to display in the pagination component.
+ *
+ * @param {integer} pagesToDisplay Maximum number of pages to display in the pagination component.
+ * @param {integer} currentPage Page currently visible.
+ * @param {integer} totalPages Total pages available.
+ * @return {object} Object containing the min and max index to display in the pagination component.
+ */
+export const getIndexes = ( pagesToDisplay, currentPage, totalPages ) => {
+	const extraPagesToDisplay = pagesToDisplay - 1;
```

Comment:
We should make this optional e.g. how many to show before/after current page.

If we are not having first/last buttons and instead having next/previous, we should perhaps add an option to control how many pages are shown at the 'edges' (naming from https://flaviusmatis.github.io/simplePagination.js/) so you can still jump to last page if needed. Combine that with a limit e.g. show no more than X page numbers.

Reply:
> We should make this optional e.g. how many to show before/after current page.

That's what the `pagesToDisplay` prop is for. From your example, I think it would be the same as `displayedPages`.

> If we are not having first/last buttons and instead having next/previous, we should perhaps add an option to control how many pages are shown at the 'edges' (naming from https://flaviusmatis.github.io/simplePagination.js/) so you can still jump to last page if needed. Combine that with a limit e.g. show no more than X page numbers.

My initial thought looking at Figma designs was that this was the purpose of the arrows. But looking at some pagination libraries like the one you linked, it's true they use the arrows for previous/next links, so we might want to change that.

Tagging @jwold for his input on this. **tl;dr:**
* What should the arrows do when clicked: link to the previous/next page or the first/last?
* In case they link to the previous/next pages, how should we add a link to the first/last?

Reply:
Prev/next is also currency core behaviour fwiw

I just thought, maybe we should actually mirror this (if it does not already exist?) https://codex.wordpress.org/Function_Reference/paginate_links

Reply:
Sounds good. :+1:

Reply:
Fixed in 0295db0d9c895299d56c3ada810546f932e79e4d.

Reply:
Looks like discussion was resolved here, but feel free to reach out if anymore questions! I think the suggestion was to make arrows go to "prev/next", agree with that. 

And also, I think we're having the first/last pages available via showing their numbers, makes sense!